app/javascript/components/student/editor/TestRunSummary.tsx

```diff
@@ -82,22 +78,14 @@ export function TestRunSummary({
   ])
 
   useEffect(() => {
-    fetchJSON(submission.links.testRun, {
-      method: 'GET',
-    }).then((json: any) => {
-      setTestRun(typecheck<TestRun>(camelizeKeys(json), 'testRun'))
-    })
-  }, [submission.uuid])
-
-  useEffect(() => {
-    channel.current = new TestRunChannel(submission, (testRun: TestRun) => {
+    channel.current = new TestRunChannel(testRun, (testRun: TestRun) => {
```

Comment:
So much cleaner!

Reply:
I do find the re-using of the `testRun` name for both the argument and the lambda parameter slightly confusing. Maybe you could use a different name for the lambda parameter?

Reply:
Agree. I renamed it to `updatedTestRun`.